src/VisualizationTests/HelixImageComparisonTests.cs

```diff
@@ -0,0 +1,428 @@
+            var rtb = view.RenderBitmap();
+            view.RenderHost.Resize((int)w, (int)h);
+            return rtb;
+        }
+
+        private static void SaveBitMapSourceAsPNG(string filePath, BitmapSource bitmapSource)
+        {
+            using (var fileStream = new FileStream(filePath, FileMode.Create))
+            {
+                BitmapEncoder encoder = new PngBitmapEncoder();
+                encoder.Frames.Add(BitmapFrame.Create(bitmapSource));
+                encoder.Save(fileStream);
+            }
+        }
+
+        private System.Drawing.Bitmap BitmapFromSource(BitmapSource bitmapsource)
+        {
+            System.Drawing.Bitmap bitmap;
+            using (MemoryStream outStream = new MemoryStream())
+            {
+                BitmapEncoder enc = new PngBitmapEncoder();
+                enc.Frames.Add(BitmapFrame.Create(bitmapsource));
+                enc.Save(outStream);
+                bitmap = new System.Drawing.Bitmap(outStream);
+            }
+            return bitmap;
+        }
+
+        private string generateTestDataPathFromTest(string testname)
+        {
+             var fileName = testname+".png";
+             string relativePath = Path.Combine(
+                GetTestDirectory(ExecutingDirectory),
+                string.Format(@"core\visualization\imageComparison\referenceImages\{0}", fileName));
+            return relativePath;
+        }
+
+        //TODO consider something like:
+        //(diferentPixelsCount / (mainImage.width* mainImage.height))*100
+        //for a percent image diff.
+        private static void compareImageColors(Bitmap image1,Bitmap image2)
+        {
+            Assert.AreEqual(image1.Width, image2.Width);
+            Assert.AreEqual(image1.Height, image2.Height);
+            Assert.AreEqual(image1.PixelFormat, image2.PixelFormat);
+
+            for (var x = 0; x < image1.Width; x++)
+            {
+                for (var y = 0; y < image1.Height; y++)
+                {
+                    var expectedCol = image1.GetPixel(x, y);
+                    var otherCol = image2.GetPixel(x, y);
+                    Assert.AreEqual(expectedCol, otherCol,$"pixel {x}:{y} was not as expected");
+                }
+            }
+        }
+
+        #endregion
+
+        #region meshes
+        [Test]
+        public void StandardMeshGeometryRender()
+        {
+            OpenVisualizationTest(@"imageComparison\spherecolors.dyn");
+            RunCurrentModel();
+            var path = generateTestDataPathFromTest(MethodBase.GetCurrentMethod().Name);
+            var bitmapsource = DynamoRenderBitmap(BackgroundPreview.View, 1024, 1024);
+#if UPDATEIMAGEDATA
+            UpdateTestData(path, bitmapsource);
+#endif
+            var refimage = new BitmapImage(new Uri(path));
+            var refbitmap = BitmapFromSource(refimage);
+            var newImage = BitmapFromSource(bitmapsource);
+
+            compareImageColors(refbitmap, newImage);
```

Comment:
Hi @mjkkirschner Given this function will always be called at the end of each test, would it be possible that we put it as part of test tear down? Just curious if we can reduce some code and ease the writing..

Reply:
hmm... I'm not sure that will be clear to those who are not well aquatinted with Nunit - also different Nunit versions behave differently, so I'd be hesitant to try that - not even sure you can fail a test in a teardown but I can look.

Maybe I can just simplify the comparison a bit more into a single utility function so theres less repeated code.